docs/gen-ai/CausalLMPipeline.md

```diff
@@ -0,0 +1,118 @@
+# What is a causal language model pipeline?
+
+The causal language model pipeline is a utility class which wraps a tokenizer and a causal language model and provides a uniformed interface for various decoding method to generate text. The pipeline is designed to be easy to use and requires only a few lines of code to generate text.
+
+In Microsoft.ML.GenAI, we will provide a generic `CausalLMPipeline` class plus a typed `CausalLMPipeline` class which specifies the type parameters for the tokenizer and the causal language model. The typed `CausalLMPipeline` class make it easier to develop consuming method for semantic kernel. see [here](./Usage.md#consume-model-from-semantic-kernel) for more details
+# Contract
+```C#
+public abstract class CausalLMPipeline
+{
+    public virtual (
+        Tensor, // output token ids [batch_size, sequence_length]
+        Tensor // output logits [batch_size, sequence_length, vocab_size]
```

Comment:
Is this the TorchSharp tensor type? Can / should it become the System.Numerics.Tensors one?

Reply:
> Is this the TorchSharp tensor type

Yes

> should it become the System.Numerics.Tensors one?
If System.Numerics.Tensors can work with libtorch, then maybe yes.

Reply:
We're some ways from having Tensor<T> integrated with TorchSharp, it is still experimental and subject to change.

Reply:
And it would be the TorchSharp shim implementing ITensor<T>, not Tensor<T> from SNT.

Reply:
Presumably it would be `ITensor<T>` such that a `Tensor<T>` could be passed in? The implementation could type test if it were a TorchSharp implementation and act accordingly, presumably.

Reply:
Yes, in the fullness of time.